main.go

```diff
@@ -6,27 +6,113 @@ import (
 	"log"
 	"net/http"
 	"regexp"
+	"strings"
 )
 
 func main() {
-	// Generate a new word for hangman
+	willRepeat := true
+	for willRepeat {
+		var (
+			word         = generateWord()
+			obscuredWord = obscureWord(word)
+			userInput    = ""
+			lives        = 6
+		)
+
+		fmt.Println("Welcome to Hangman!")
+		fmt.Printf("You have %d lives to guess the word. \n", lives)
+
+		for lives >= 0 {
+			if lives == 0 {
+				loseGame(word, &willRepeat)
+				break
+			} else if obscuredWord == word {
+				winGame(word, &willRepeat)
+				break
+			}
+
+			fmt.Printf("Lives: %d. Word: %s\n", lives, obscuredWord)
```

Comment:
nit: same as above, no need to use printf if we're going to be on a new line anyways